File: 07_User_Interface/app/src/main/java/com/a07_user_interface/MainActivity.java
package com.a07_user_interface;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity  {
    int light_id = 2;
    boolean isGreen = true;
    boolean flag_plus = false;

    @Override
    protected void onCreate(Bundle savedInstanceState)  {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView redLight = findViewById(R.id.RedLight);
        TextView yellowLight = findViewById(R.id.YellowLight);
        TextView greenLight = findViewById(R.id.GreenLight);
        ImageView Human = findViewById(R.id.Human);

        TextView[] lights = {redLight, yellowLight, greenLight};
        int[]  lightsSource = {R.drawable.red_light, R.drawable.yellow_light, R.drawable.green_light};
        Thread thread = new Thread(() -> {
            while (true)  {
                if (light_id == 2) {
                    if (isGreen) {
                        Animation forwardAnim = AnimationUtils.loadAnimation(this, R.anim.human_go_forward);
                        Human.startAnimation(forwardAnim);
                    } else {
                        Animation backwardAnim = AnimationUtils.loadAnimation(this, R.anim.human_go_backward);
                        Human.startAnimation(backwardAnim);
                    }
                    flag_plus = false;
                } else if (light_id == 0) {
                    flag_plus = true;
                }
                lights[light_id].setBackgroundResource(lightsSource[light_id]);
                if (light_id == 0) {
                    if (isGreen) {
                        Human.setRotationY(180);
                        isGreen = false;
                    }
                    else {
                        Human.setRotationY(0);
                        isGreen = true;
                    }
                }
                try {
                    Thread.sleep(1000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
                lights[light_id].setBackgroundResource(R.drawable.circle);

                if (flag_plus)
                    light_id++;
                else light_id--;
            }
        });
        thread.start();


        Button bNextPage = findViewById(R.id.NextPage);
        bNextPage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, TextMotion.class);
                startActivity(intent);
            }
        });
    }
}

File: 05_Frame_Stack/app/src/main/java/com/example/a05_frame_stack/MainActivity.kt
package com.example.a05_frame_stack

import android.annotation.SuppressLint
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.ConstraintSet
import androidx.core.graphics.BlendModeColorFilterCompat
import androidx.core.graphics.BlendModeCompat
import androidx.core.view.get
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator

var count = 0

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val forwardButton = findViewById<Button>(R.id.Forward)
        val backwardButton = findViewById<Button>(R.id.Backward)
        val nextPageButton = findViewById<Button>(R.id.NextPage)
        val stackDepthInfo = findViewById<TextView>(R.id.StackDepth)

        var stackDepth = intent.getIntExtra("stackDepth", 0)
        stackDepthInfo.text = "Stack Depth: " + stackDepth.toString()
        backwardButton.setOnClickListener {
            if (stackDepth != 0) {
                val intent = Intent(this@MainActivity, MainActivity::class.java)
                stackDepth--
                intent.putExtra("stackDepth", stackDepth)
                startActivity(intent)
            }
        }
        forwardButton.setOnClickListener {
            val intent = Intent(this@MainActivity, MainActivity::class.java)
            stackDepth++
            intent.putExtra("stackDepth", stackDepth)
            startActivity(intent)
        }
        nextPageButton.setOnClickListener {
            val intent = Intent(this@MainActivity, AttachPageActivity::class.java)
            startActivity(intent)
        }

//        val mainTab = tabs.newTab()
//        mainTab.setText("Main")
//        tabs.addTab(mainTab, true)
//
//        returnButton.visibility = View.INVISIBLE
//        val newTab = tabs.newTab()
//        newTab.setText("Attached Page")
//        newTab.setTabLabelVisibility(TabLayout.TAB_LABEL_VISIBILITY_UNLABELED)
//        tabs.addTab(newTab, false)
//
//
//        // Вкладки
//        attachButton.setOnClickListener {
//            newTab.setTabLabelVisibility(TabLayout.TAB_LABEL_VISIBILITY_LABELED)
//            newTab.select()
//            forwardButton.visibility = View.INVISIBLE
//            backwardButton.visibility = View.INVISIBLE
//            stackDepthInfo.visibility = View.INVISIBLE
//            detachButton.visibility = View.INVISIBLE
//            attachButton.visibility = View.INVISIBLE
//            nextPageButton.visibility = View.INVISIBLE
//            returnButton.visibility = View.VISIBLE
//        }
//        detachButton.setOnClickListener {
//            newTab.setTabLabelVisibility(TabLayout.TAB_LABEL_VISIBILITY_UNLABELED)
//            newTab
//        }
//        returnButton.setOnClickListener {
//            mainTab.select()
//            forwardButton.visibility = View.VISIBLE
//            backwardButton.visibility = View.VISIBLE
//            stackDepthInfo.visibility = View.VISIBLE
//            detachButton.visibility = View.VISIBLE
//            attachButton.visibility = View.VISIBLE
//            nextPageButton.visibility = View.VISIBLE
//            returnButton.visibility = View.INVISIBLE
//        }
//
//        var stackDepth = intent.getIntExtra("stackDepth", 0)
//        stackDepthInfo.text = "Stack Depth: " + stackDepth.toString()
//        backwardButton.setOnClickListener {
//            if (stackDepth != 0) {
//                val intent = Intent(this@MainActivity, MainActivity::class.java)
//                stackDepth--
//                intent.putExtra("stackDepth", stackDepth)
//                startActivity(intent)
//            }
//        }
//        forwardButton.setOnClickListener {
//            val intent = Intent(this@MainActivity, MainActivity::class.java)
//            stackDepth++
//            intent.putExtra("stackDepth", stackDepth)
//            startActivity(intent)
//        }
//        nextPageButton.setOnClickListener {
//            val intent = Intent(this@MainActivity, DialogActivity::class.java)
//            startActivity(intent)
//        }
    }
}